### ISSUE TITLE
Add gaussian blur to background of the pause menu

### ISSUE THREAD

**Issue by RaptorParkowsky** [colobot/colobot#656]
Well, not really necessary feature and not even possible for now, as it probably require postprocessing shaders and will be implemented after the refactoring of graphic or even also UI engine, but I think it would look cool and refreshed at least.

Here's previsualization concept:

Original:
![schowek01](https://cloud.githubusercontent.com/assets/4019345/10144963/b942f5a0-6620-11e5-81ca-d774152fd3e9.png)

Postprocessed:
![schowek02](https://cloud.githubusercontent.com/assets/4019345/10144978/c9ad6fc4-6620-11e5-9d58-8f56921bbc82.png)

* Note that sparks particles in UI were only disabled to make this concept photo.

**Comment by krzys-h** [comment]
:+1:

**Comment by tomaszkax86** [comment]
This can be done without shaders. One simply needs to render the scene (to offscreen buffer if possible), read it and post-process it on CPU, then render it on screen. Optionally, we can use mipmap generation to make it cheaper. Since the scene doesn't change at all, this needs to be done once.

**Comment by tomangelo2** [comment]
Won't CPU processing cause lag while opening pause menu?

**Comment by RaptorParkowsky** [comment]
@tomangelo2 That's what I was afraid of. Well, I thought that would be postprocessed rather by GPU, no CPU. For example generating gaussian blur with strenght of x=15px, y=15px in GIMP is really slow, even if image is in 800x600 resolution (at least in my old compie...). I guess in-game that would cause even 5 seconds of lag.

Another thing is that we shouldn't always "make graphic miracles" on CPU, as this is really not designed for it.

@tomaszkax86 How it would look like with mipmap generation? Well, I really hope it will be in quality like other games have, especially if we will decide to use that effect somewhere else, not only in pause menu and use smooth transition for it.

**Comment by Ghoughpteighbteau** [comment]
gimp does a little more work than is necessary, due to bit depth and gamma concerns, an 800x600 image would not be prohibitively expensive to blur at 15 pixels. it's 60 samples per pixel to compute the blur function. As long as you're smart about it, it's pretty easy to do. You can compute the image in the background and fade it in.

However, you don't need to do a perfectly accurate Gaussian blur on a 800x600 image (which is what gimp is going on top of everything else), you can downsample your target image severely and get very good results. 

A guassian blur on a 200x150 image will produce very passable results and a 64* reduction in render time. 

Basically, if you want a static blur in the background at 800x600 and are tolerant of downsamples and compute it in the background, very reasonable to do. If you want live perfectly accurate guassian blur at 4K resolution,  I don't think modern GPU's could hack it.

reference:
![no filter](http://i.imgur.com/V24rtiT.png)

accurate 15px guassian blur:
![15gb](http://i.imgur.com/UK84vBY.png)

downsampled 4xp guassian blur bilinear interpolation:
![4gb4s](http://i.imgur.com/Q5H171N.png)

**Comment by tomaszkax86** [comment]
Mipmap generation done by OpenGL is a somewhat easy and fast way to downsample image automatically. If you want, I can make some tests before we implement one method.

**Comment by krzys-h** [comment]
Enabled by default after 8a17bc901c8ea526278287548bf687640c8a5973
